File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Par_ExportCauService/Models/Par_ExportCauModel.cs
using nvxapp.server.data.Entities.Tenant.GestionePresenze;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_ExportCau_CausaliService.Models;
using nvxapp.server.service.ClientServer_Service.ModelsBase;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_ExportCauService.Models
{
    public class Par_ExportCauModel
    {
        public int Id { get; set; }
        public int IdAz_Anagrafica { get; set; }
        public string Codice { get; set; } = string.Empty;
        public string Descrizione { get; set; } = string.Empty;
        public Par_Export_TipoFile TipoFile { get; set; }
    }

    public class Par_ExportCau_GetAll_InModel { }
    public class Par_ExportCau_GetAll_OutModel : ModelResult
    {
        public List<Par_ExportCauModel> Par_ExportCau { get; set; } = new();
    }

    public class Par_ExportCau_Get_InModel
    {
        public int Id { get; set; }
    }
    public class Par_ExportCau_Get_OutModel : ModelResult
    {
        public Par_ExportCauModel? Par_ExportCau { get; set; }
        public List<Par_ExportCau_CausaliModel> Par_ExportCau_Causali { get; set; } = new();
    }

    public class Par_ExportCau_Put_InModel
    {
        public Par_ExportCauModel Par_ExportCau { get; set; } = new();
        public List<Par_ExportCau_CausaliModel> Par_ExportCau_Causali { get; set; } = new();
    }
    public class Par_ExportCau_Put_OutModel : ModelResult
    {
        public Par_ExportCauModel? Par_ExportCau { get; set; }
        public List<Par_ExportCau_CausaliModel> Par_ExportCau_Causali { get; set; } = new();
    }

    public class Par_ExportCau_Delete_InModel
    {
        public int Id { get; set; }
    }
    public class Par_ExportCau_Delete_OutModel : ModelResult { }
}

File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Par_ExportCau/Models/par-export-cau-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";
import { Par_ExportCau_CausaliModel } from "../../Par_ExportCau_Causali/Models/par-export-cau-causali-model";

export class Par_ExportCauModel {
  id: number;
  idAz_Anagrafica: number;
  codice: string;
  descrizione: string;
  tipoFile: Par_Export_TipoFile;
}

export enum Par_Export_TipoFile {
  CSV = 0,
  TXT = 1
}

export class Par_ExportCau_GetAll_InModel { }
export class Par_ExportCau_GetAll_OutModel extends ModelResult {
  par_ExportCau: Par_ExportCauModel[];
  constructor() {
    super();
    this.par_ExportCau = [];
  }
}

export class Par_ExportCau_Get_InModel {
  id: number = 0;
}
export class Par_ExportCau_Get_OutModel extends ModelResult {
  par_ExportCau: Par_ExportCauModel;
  par_ExportCau_Causali: Par_ExportCau_CausaliModel[];
  constructor() {
    super();
    this.par_ExportCau = new Par_ExportCauModel();
    this.par_ExportCau_Causali = [];
  }
}

export class Par_ExportCau_Put_InModel {
  par_ExportCau: Par_ExportCauModel;
  par_ExportCau_Causali: Par_ExportCau_CausaliModel[];
  constructor() {
    this.par_ExportCau = new Par_ExportCauModel();
    this.par_ExportCau_Causali = [];
  }
}
export class Par_ExportCau_Put_OutModel extends ModelResult {
  par_ExportCau: Par_ExportCauModel;
  par_ExportCau_Causali: Par_ExportCau_CausaliModel[];
  constructor() {
    super();
    this.par_ExportCau = new Par_ExportCauModel();
    this.par_ExportCau_Causali = [];
  }
}

export class Par_ExportCau_Delete_InModel {
  id: number = 0;
}
export class Par_ExportCau_Delete_OutModel extends ModelResult { }
